src/client/lib/chess/piece.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import Bishop from './bishop';
+import King from './king';
+import Knight from './knight';
+import Pawn from './pawn';
+import Queen from './queen';
+import Rook from './rook';
+const ANIMATION_DURATION = 750;
+
+/*
+ * piece
+ *
+ * Component for every chess piece. It animates any change in x and y
+ * using in out cubic easing. Its SVG elements uses a space of a square width
+ * side 1.
+ *
+ * Props:
+ *   x       - X coordinate on SVG space.
+ *   y       - Y coordinate on SVG space.
+ *   type    - Rook (r), pawn (p), king (k) etc. Uses Chess.js convention.
+ *   color   - Whether 'light' or 'dark' piece.
+ *   onClick - Callback to be called when clicked, with X and Y parameters.
+ *
+ * Usage:
+ *
+ * <svg viewBox={'0 0 1 1'}>
+ *   <ChessPiece type='k' color='light' x={0} y={0}
+ *               onClick={onClickFn}/>
+ * </svg>
+ */
+class ChessPiece extends React.Component {
+  /**
+  * Sets the x and y of the state on creation.
+  */
+  componentWillMount() {
+    this.setState({
+      x: this.props.x,
+      y: this.props.y
+    });
+  }
+
+  /**
+  * If there is a change in x and y, saves old X, Y, and current time. Starts
+  * animation.
+  */
+  componentWillReceiveProps (nextProps) {
```

Comment:
No space before the paren just to be consistent with the code style elsewhere. Can you also add https://eslint.org/docs/rules/space-before-function-paren to the .eslintrc file?